lib/scholar/interpolation/cubic_spline.ex

```diff
@@ -0,0 +1,147 @@
+defmodule Scholar.Interpolation.CubicSpline do
+  @moduledoc """
+  Cubic spline interpolation
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:coefficients, :x]}
+  defstruct [:coefficients, :x]
+
+  opts = [
+    extrapolate: [
+      required: false,
+      default: true,
+      type: :boolean,
+      doc: "if false, out-of-bounds x values raise."
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts)
+
+  @doc """
+  Fits a cubic spline interpolation of the given `(x, y)` points
+  """
+  defn train(x, y) do
```

Comment:
We should document that x must be in increasing order.

Reply:
We can apply sorting with argsort to deal with this, I think

Reply:
Sounds good! In most cases `train` can be expected to be somewhat expensive, so an additional sort doesn't hurt and will prevent from bugs.